cobalt-parser/src/parser/expr/mod.rs

```diff
@@ -0,0 +1,1041 @@
+                    errors,
+                );
+            }
+        }
+
+        // Handle the else.
+        // We know the current token is an `else` keyword.
+
+        self.next();
+        if self.current_token.is_none() {
+            errors.push(CobaltError::ExpectedFound {
+                ex: "block expression",
+                found: ParserFound::Eof,
+                loc: self.current_token.unwrap().span,
+            });
+            return (
+                Box::new(IfAST::new(
+                    span,
+                    cond,
+                    if_true,
+                    Box::new(NullAST::new(SourceSpan::from((0, 1)))),
+                )),
+                errors,
+            );
+        }
+
+        let (if_false, if_false_errors) = self.parse_block_expr();
+        errors.extend(if_false_errors);
+
+        (Box::new(IfAST::new(span, cond, if_true, if_false)), errors)
+    }
+
+    /// Going into this function, `current_token` is assumed to be an `@`.
+    /// Instrinsics can be functions; these are parsed in the function call
+    /// parsing methods.
+    ///
+    /// ```text
+    /// instinsic := '@' [ident | keyword]
+    /// ```
+    pub(crate) fn parse_intrinsic(&mut self) -> (BoxedAST<'src>, Vec<CobaltError<'src>>) {
+        assert!(self.current_token.is_some());
+        assert_eq!(self.current_token.unwrap().kind, TokenKind::At);
+
+        let span = self.current_token.unwrap().span;
+        let mut errors = vec![];
+        self.next();
+
+        // ---
+
+        if self.current_token.is_none() {
+            errors.push(CobaltError::ExpectedFound {
+                ex: "name of intrinsic",
+                found: ParserFound::Eof,
+                loc: self.current_token.unwrap().span,
+            });
+            return (Box::new(ErrorAST::new(self.source.len().into())), errors);
+        }
+
+        let name: Cow<'_, str> = match self.current_token.unwrap().kind {
+            TokenKind::Ident(ident) => Cow::Borrowed(ident),
+            TokenKind::Keyword(kw) => Cow::Borrowed(kw.as_str()),
+            _ => {
+                errors.push(CobaltError::ExpectedFound {
+                    ex: "name of intrinsic",
+                    found: ParserFound::Str(self.current_token.unwrap().kind.to_string()),
+                    loc: self.current_token.unwrap().span,
+                });
+                return (
+                    Box::new(ErrorAST::new(self.current_token.unwrap().span)),
+                    errors,
+                );
+            }
+        };
+
+        self.next();
+
+        // ---
+
+        (Box::new(IntrinsicAST::new(span, name)), errors)
+    }
+
+    pub(crate) fn check_fn_call(&mut self) -> bool {
+        if self.current_token.is_none() {
+            return false;
+        }
+
+        let idx_on_entry = self.cursor.index;
+
+        match self.current_token.unwrap().kind {
+            TokenKind::Ident(_) => {
+                self.next();
+            }
```

Comment:
Any expression can be called, not just variables and intrinsics

Reply:
For example?

Reply:
Fixed in [fb46544](https://github.com/cobalt-language/cobalt-lang/pull/115/commits/fb4654400a93e33ed4e001a5a38551f970581aa8)